.gitignore

```diff
@@ -1,19 +1,18 @@
 *.iml
 .gradle
 /local.properties
-/.idea/caches
-/.idea/libraries
-/.idea/modules.xml
-/.idea/workspace.xml
-/.idea/navEditor.xml
-/.idea/assetWizardSettings.xml
+**/.idea/**
+!**/.idea/codeStyleSettings.xml
```

Comment:
what does the "!" do in gitignore?

Reply:
It's supposed to mean "don't ignore". It's paired with the wildcard ignore on the line above. Essentially instead of having to add each file from `.idea/` that we want to ignore, we can say "ignore everything in `.idea/` *except* for these files.